src/plays/typing-speed-test/components/Main.js

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState, useRef } from "react";
+import { VscRefresh } from "react-icons/vsc";
+
+// Project local imports
+import { generateWords } from "../utils";
+import Stats from "./Stats";
+import Timer from "./Timer";
+import Word from "./Word";
+import ResultModal from "./ResultModal";
+
+const Main = () => {
+  const userInputRef = useRef(null);
+  const [words, setWords] = useState("");
+  const [userInput, setUserInput] = useState("");
+  const [timer, setTimer] = useState(60);
+  const [status, setStatus] = useState("waiting");
+  const [activeWordIndex, setActiveWordIndex] = useState(0);
+  const [typedWordsArray, setTypedWordsArray] = useState([]);
+  const [incorrectWords, setIncorrectWords] = useState(0);
+  const [correctWords, setCorrectWords] = useState(0);
+  const [correctCharLength, setCorrectCharLength] = useState(0);

```

Comment:
Check if we can combine these states into one state object.

Reply:
Done